### DISCUSSION TITLE
Reintroducing automatic note creation through `:ObsidianNew` as an option

### DISCUSSION THREAD

**Discussion by Vourliotis** [epwalsh/obsidian.nvim#600] [Ideas]
Hey everyone,

First off, thanks a lot to everyone involved in this plugin! I wanted to start a discussion about a "feature" that used to be in the plugin but was removed in #490.

`:ObsidianNew` automatically created any non-existent folders given a path, along with the new note. When creating a new note, I often include a path with folders that might not exist yet. Without the automatic creation, an error gets thrown due to a function trying to access a non-existent path (`get_workspace_for_dir`).

I tried handling this with the `enter_note` callback, but since another function or callback runs first, I wasn't successful. The only way to make it work again was through an autocmd:

```lua
vim.api.nvim_create_autocmd('BufEnter', {
  pattern = '*.md',
  callback = function()
    local bufname = vim.api.nvim_buf_get_name(0)
    local dirname = vim.fn.fnamemodify(bufname, ':h')

    vim.fn.mkdir(dirname, 'p')
    vim.api.nvim_command('write')
  end,
})

```

Do you think it’s worth bringing this feature back as an option? Are there any good workarounds for this that I might have missed?

Would love to hear your thoughts!

**Comment by epwalsh** [comment]
Hey @Vourliotis I would consider this a bug. Let me see if I can fix.

**Reply by epwalsh** [reply]
Ok, so I decided the best course of action was the always ensure the parent directory exists when we create a new note, even if we're not writing the note itself to disk. I implemented this in 49cfe4d which will be in the next release. Let me know if you still have issues after upgrading.

**Comment by Vourliotis** [comment]
Seems to work flawlessly @epwalsh, thank you for the change!
